Backend/routes/notifications.js

```diff
@@ -47,12 +46,40 @@ router.get("/notifications", verifyToken, async (req, res) => {
       orderBy: { createdAt: "desc" },
     });
 
-    const formattedNotifications = notifications.map((notification) => ({
-      ...notification,
-      timeAgo: formatDistanceToNow(new Date(notification.createdAt), {
-        addSuffix: true,
-      }),
-    }));
+    const groupedNotifications = notifications.reduce((acc, notif) => {
+      const key = `${notif.type}-${notif.listingId || ""}`;
+      if (!acc[key]) {
+        acc[key] = [];
+      }
+      acc[key].push(notif);
+      return acc;
+    }, {});
+
+    const formattedNotifications = Object.values(groupedNotifications).map(
+      (group) => {
+        if (group.length > 1 && group[0].type === "LIKE") {
```

Comment:
what about the other notif types?

Reply:
i only wanted to do the notification batching for likes because those are the ones that tend to grow really fast within a short amount of time, but i could also implement that in the follow if thats better?